### PR TITLE
ARCH-5766: Add `SingleAsync` and `SingleOrDefaultAsync` QueryEndpoint methods

### PR DESCRIPTION
**Pull request by SergeiPavlov**

### CODE REVIEW

File: Orm/Xtensive.Orm/Orm/QueryEndpoint.cs [hunk 1]
```diff
@@ -265,12 +265,34 @@ public IQueryable<FullTextMatch<T>> ContainsTable<T>([NotNull] Expression<Func<C
...
+    /// <returns>
+    /// The <see cref="Entity"/> specified <paramref name="key"/> identifies.
+    /// </returns>
+    /// <exception cref="KeyNotFoundException">Entity with the specified key is not found.</exception>
+    public async ValueTask<Entity> SingleAsync(Key key, CancellationToken ct = default)
+    {
+      if (key is null) {
+        return null;
+      }
+      var result = await SingleOrDefaultAsync(key, ct);
```
Comment (botinko) [inline Orm/Xtensive.Orm/Orm/QueryEndpoint.cs]: `ConfigureAwait(false)` for compatibility with DO code standards.

File: Orm/Xtensive.Orm/Orm/QueryEndpoint.cs [hunk 2]
```diff
@@ -283,37 +305,73 @@ public Entity Single(Key key)
...
-        if (state==null || state.IsNotAvailableOrMarkedAsRemoved
-          || !key.TypeReference.Type.UnderlyingType.IsAssignableFrom(state.Type.UnderlyingType))
-          // No state or Tuple = null or incorrect query type => no data in storage
-          result = null;
-        else
-          result = state.Entity;
-
-        tx.Complete();
+
+        state = await session.Handler.FetchEntityStateAsync(key, ct);
```
Comment (botinko) [inline Orm/Xtensive.Orm/Orm/QueryEndpoint.cs]: `ConfigureAwait(false)` for compatibility with DO code standards.

File: Orm/Xtensive.Orm/Orm/Providers/SqlSessionHandler.Fetching.cs [hunk 3]
```diff
@@ -57,7 +57,20 @@ public override EntityState FetchEntityState(Key key)
...
+    /// <summary>
+    /// Fetches an <see cref="EntityState"/>.
+    /// </summary>
+    /// <param name="key">The key.</param>
+    /// <returns>The key of fetched <see cref="EntityState"/>.</returns>
+    public override async Task<EntityState> FetchEntityStateAsync(Key key, CancellationToken ct = default)
+    {
+      var type = key.TypeReference.Type;
+      await prefetchManager.PrefetchAsync(key, type, PrefetchHelper.GetCachedDescriptorsForFieldsLoadedByDefault(Session.Domain, type), ct);
+      await prefetchManager.ExecuteTasksAsync(true, ct);
```
Comment (botinko) [inline Orm/Xtensive.Orm/Orm/Providers/SqlSessionHandler.Fetching.cs:70]: `ConfigureAwait(false)` for compatibility with DO code standards.

**Review by botinko — COMMENTED** [review]
LGTM

File: Orm/Xtensive.Orm/Orm/QueryEndpoint.cs (hunk 1 above)
Comment (SergeiPavlov) [reply Orm/Xtensive.Orm/Orm/QueryEndpoint.cs]: done

File: Orm/Xtensive.Orm/Orm/QueryEndpoint.cs (hunk 2 above)
Comment (SergeiPavlov) [reply Orm/Xtensive.Orm/Orm/QueryEndpoint.cs]: done

File: Orm/Xtensive.Orm/Orm/Providers/SqlSessionHandler.Fetching.cs (hunk 3 above)
Comment (SergeiPavlov) [reply Orm/Xtensive.Orm/Orm/Providers/SqlSessionHandler.Fetching.cs:70]: done

**Pull request merged**